### DISCUSSION TITLE
Blazor, UseCsla()

### DISCUSSION THREAD

**Discussion by JurjendeGroot** [MarimerLLC/csla#2210] [Questions]
Hello,

I'm in a project using the ABP.IO framework with Blazor.
I'm using Csla 5.4.1

Upon Configuring the dataportal:

```c#
app.UseCsla((config) =>
              config.DataPortal().DefaultProxy(
                typeof(Csla.DataPortalClient.HttpProxy), "https://localhost:44323/api/Blazor/DataPortal") );
```

 I get a red squiggle under 'app.UseCsla' after wich the tooltip is telling me that:

```
CS1705: Assembly 'Csla.AspNetCore' with identity 'Csla.AspNetCore, Version 5.4.1.0, Culture.....' uses 'Microsoft.AspNetCore.Http.Abstractions, Version=5.0.0.0, Culture=.....' wich has a higher version than referenced assembly 'Microsoft.AspNetCore.Http.Abstractions' with identity 'Microsoft.AspNetCore.Http.Abstractions, Version 2.2.0.0, .....'
```

I'm a little puzzled, checking nuget tells met that the current version for Microsoft.AspNetCore.Http.Abstractions is 2.2.0
https://www.nuget.org/packages/Microsoft.AspNetCore.Http.Abstractions/

I'm not sure where this is coming from. Csla.AspNetCore does have references to Microsoft packages with versions 5.0.0.0 or higher, but Microsoft.AspNetCore.Http.Abstractions doesn't seem to be one of them ...
https://www.nuget.org/packages/Csla.AspNetCore/5.4.1

Am I missing something here, any help is greatly appreciated.

Jurjen.

**Comment by rockfordlhotka** [comment]
Is your project targeting .NET 5 or 3.1?

Right now (in CSLA 5.4) the .NET 5 `Csla.dll` references all the latest .NET 5 dependencies. The netstandard2.0 `Csla.dll` references the latest .NET 3.x dependencies.

**Reply by JurjendeGroot** [reply]
It is targeting .NET 5.0

**Reply by rockfordlhotka** [reply]
Hmm.

The [Csla.AspNetCore](https://github.com/MarimerLLC/csla/blob/main/Source/Csla.AspNetCore.NetCore3.1/Csla.AspNetCore.NetCore3.1.csproj) project doesn't reference any external dependencies directly other than the Microsoft app package:

[FrameworkReference Include="Microsoft.AspNetCore.App"](https://github.com/MarimerLLC/csla/blob/a87cce9253694570222d5ed78c99fcdae00a0cf8/Source/Csla.AspNetCore.NetCore3.1/Csla.AspNetCore.NetCore3.1.csproj#L44)

As I understand it, that should be automatic to the latest version, which was probably 5.0.0 at the time that version of CSLA was released.

**Reply by JurjendeGroot** [reply]
At this time I seem to have solved it using:

```c#
private void ConfigureCsla(WebAssemblyHostBuilder builder, ServiceConfigurationContext context)
{
            context.Services.AddCsla(
                                (config) => config.DataPortal().DefaultProxy(
                                    typeof(myHttpProxy), "https://localhost:44321/api/Blazor/DataPortal"))
                            ;

            //builder.UseCsla((config) =>
            //      config.DataPortal().DefaultProxy(
            //        typeof(myHttpProxy), "https://localhost:44323/api/Blazor/DataPortal")
            //      );
}
```

I'm still testing but this seems to work.
